crates/oxc_cli/src/command.rs

```diff
@@ -43,96 +41,73 @@ pub struct LintCommand {
-    #[bpaf(argument("NUMBER"), hide_usage)]
-    pub max_warnings: Option<usize>,
+    /// list all the rules that are currently registered
+    #[bpaf(switch, hide_usage)]
+    pub rules: bool,
 
     /// Number of threads to use. Set to 1 for using only 1 CPU core.
-    #[bpaf(argument("NUMBER"), hide_usage)]
+    #[bpaf(argument("INT"), hide_usage)]
     pub threads: Option<usize>,
 }
 
-#[derive(Debug, Clone, Bpaf)]
-pub struct WalkOptions {
-    /// Disables excluding of files from .eslintignore files
-    #[bpaf(switch)]
-    pub no_ignore: bool,
-
-    /// Specify the file to use as your .eslintignore
-    #[bpaf(argument("PATH"), fallback(".eslintignore".into()))]
-    pub ignore_path: OsString,
-
-    /// Specify patterns of files to ignore (in addition to those in .eslintignore)
-    #[bpaf(argument("PATTERN"), many)]
-    pub ignore_pattern: Vec<String>,
-
-    /// Single file, single path or list of paths
-    #[bpaf(positional("PATH"), many)]
-    pub paths: Vec<PathBuf>,
-}
-
-static FILTER_HELP: &str = r#"
-To allow or deny a rule, use multiple -A <NAME> or -D <NAME>.
-
-For example "-D correctness -A no-debugger" or "-A all -D no-debugger".
-
-The categories are:
-  * correctness - code that is outright wrong or useless
-  * suspicious  - code that is most likely wrong or useless
-  * pedantic    - lints which are rather strict or have occasional false positives
-  * style       - code that should be written in a more idiomatic way
-  * nursery     - new lints that are still under development
-  * restriction - lints which prevent the use of language and library features
-  * all         - all the categories listed above
-
-The default category is "-D correctness".
-"#;
-
 #[derive(Debug, Clone, Bpaf)]
 pub struct LintOptions {
-    #[bpaf(external(lint_filter), map(LintFilter::into_tuple), many, group_help(FILTER_HELP))]
+    #[bpaf(external(lint_filter), map(LintFilter::into_tuple), many)]
     pub filter: Vec<(AllowWarnDeny, String)>,
 
-    /// Fix as many issues as possible. Only unfixed issues are reported in the output
-    #[bpaf(switch)]
-    pub fix: bool,
+    #[bpaf(external)]
+    pub fix_options: FixOptions,
 
-    /// Display the execution time of each lint rule
-    #[bpaf(switch, env("TIMING"), hide_usage)]
-    pub timing: bool,
+    #[bpaf(external)]
+    pub ignore_options: IgnoreOptions,
 
-    /// list all the rules that are currently registered
-    #[bpaf(switch, hide_usage)]
-    pub rules: bool,
+    #[bpaf(external)]
+    pub warning_options: WarningOptions,
 
-    #[bpaf(external(cli_options), hide_usage)]
-    pub cli: CliOptions,
+    #[bpaf(external)]
+    pub misc_options: MiscOptions,
 
-    #[bpaf(external(walk_options), hide_usage)]
-    pub walk: WalkOptions,
+    /// Single file, single path or list of paths
+    #[bpaf(positional("PATH"), many)]
+    pub paths: Vec<PathBuf>,
```

Comment:
This is being shown at the bottom, is it possible to move the help message to the top?

Reply:
Not at the moment, at least not in combinations with other group, I need to think about it a bit more.

Reply:
By that I mean it'll take some time - a few days/week.